### PR TITLE
docs: update HttpHeader documentation

### PR DESCRIPTION
**Pull request by sreevani-ship-it**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Existing documentation is not detailed enough.
Issue Number: N/A

## What is the new behavior?
Added additional details on API calls.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jenniferfell** [comment]
Internal link: https://angular-team.atlassian.net/browse/DOCS-133

File: packages/common/http/src/headers.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ interface Update {
 }
 
 /**
- * Immutable set of Http headers, with lazy parsing.
+ * `HTTPHeaders` represents the header configuration options for an Http request.
```
Comment (CaerusKaru) [inline packages/common/http/src/headers.ts]: Use `HTTP` consistent throughout

File: packages/common/http/src/headers.ts [hunk 2]
```diff
@@ -41,6 +42,9 @@ export class HttpHeaders {
    */
   private lazyUpdate: Update[]|null = null;
 
+  /* Constructs a new HTTP header object with the given values.
```
Comment (CaerusKaru) [inline packages/common/http/src/headers.ts]: If this is a single line comment, use single line quotes:
`/** Constructs a new HTTP header object with the given values */`

File: packages/common/http/src/headers.ts [hunk 3]
```diff
@@ -81,7 +85,12 @@ export class HttpHeaders {
   }
 
   /**
-   * Checks for existence of header by given name.
+   * Checks for existence of a header by a given name.
+   *
+   * @param name The header name to check for existence.
+   *
+   * @returns True if the header exits,
```
Comment (CaerusKaru) [inline packages/common/http/src/headers.ts]: If it's a boolean, you can just say `@returns whether the the header exists`

File: packages/common/http/src/headers.ts [hunk 4]
```diff
@@ -109,22 +124,50 @@ export class HttpHeaders {
   }
 
   /**
-   * Returns list of header values for a given name.
+   * Returns a list of header values for a given header name.
+   *
+   * @param name The header name to retrieve the values from.
```
Comment (CaerusKaru) [inline packages/common/http/src/headers.ts]: Grammatically, "The header name from which to retrieve the values"

File: packages/common/http/src/headers.ts [hunk 5]
```diff
@@ -109,22 +124,50 @@ export class HttpHeaders {
...
     this.init();
 
     return this.headers.get(name.toLowerCase()) || null;
   }
 
+  /**
+   * Appends a new header value to the existing set of
+   * header values.
+   *
+   * @param name The header name to append the values to.
```
Comment (CaerusKaru) [inline packages/common/http/src/headers.ts]: Grammatically, "The header name for which to append the values"

**Review by CaerusKaru — APPROVED** [review]
Looks good, some minor nits

**Comment by IgorMinar** [comment]
@chembu this PR fails CI lint check and needs to be reformatted. Can you please fix that and push a fixup commit to this branch (or fix the commit via squash/amend and force-push). thanks

**Review by mhevery — APPROVED** [review]

**Pull request closed without merging**